src/App.js

```diff
@@ -14,286 +14,431 @@ export default App;
-    } else {
-      if (this.state.counter) {
-        newcol7.push('y');
-        console.log(`new array is${newcol7}`)
-
-        this.setState({
-          col7: newcol7,
-          counter: !this.state.counter,
-        })
-      } else {
-        newcol7.push('r');
-        console.log(`new array is${newcol7}`)
-
-        this.setState({
-          col7: newcol7,
-          counter: !this.state.counter,
-        })
-      }
-    }
+    console.log(`new array is${columnCopy}`);
+
+    this.setState({
+      columns: this.state.columns.map((column, idx) => {
+        // map through columns
+        if (idx === index) {
+          // where column === column[index] -> replace with columnCopy
+          return columnCopy;
+        } else {
+          return column; // where current column !== column[index] - copy as is
+        }
+      }),
+      counter: !this.state.counter,
+    });
   }
 
   render() {
-    console.log('rendered game')
+    console.log("rendered game");
 
     return (
       <div>
         <h2>Connect 4</h2>
 
         <div>
-          <SelectColumn onClick={i => this.handleClickCol1(i)} value={1} />
-          <SelectColumn onClick={i => this.handleClickCol2(i)} value={2} />
-          <SelectColumn onClick={i => this.handleClickCol3(i)} value={3} />
-          <SelectColumn onClick={i => this.handleClickCol4(i)} value={4} />
-          <SelectColumn onClick={i => this.handleClickCol5(i)} value={5} />
-          <SelectColumn onClick={i => this.handleClickCol6(i)} value={6} />
-          <SelectColumn onClick={i => this.handleClickCol7(i)} value={7} />
+          {this.columns.map((column, index) => (
+            <SelectColumn
+              key={index}
+              onClick={() => this.handleClickCol(index)}
+            />
+          ))}
         </div>
 
-
         <div className="counter-column-container">
-          <DisplayHoles value={this.state.col1[5]} className={this.state.col1[5] === 'y' ? "displayHolesContainerYellow" : "displayHolesContainerRed"} />
-          <DisplayHoles value={this.state.col2[5]} className={this.state.col2[5] === 'y' ? "displayHolesContainerYellow" : "displayHolesContainerRed"} />
-          <DisplayHoles value={this.state.col3[5]} className={this.state.col3[5] === 'y' ? "displayHolesContainerYellow" : "displayHolesContainerRed"} />
-          <DisplayHoles value={this.state.col4[5]} className={this.state.col4[5] === 'y' ? "displayHolesContainerYellow" : "displayHolesContainerRed"} />
-          <DisplayHoles value={this.state.col5[5]} className={this.state.col5[5] === 'y' ? "displayHolesContainerYellow" : "displayHolesContainerRed"} />
-          <DisplayHoles value={this.state.col6[5]} className={this.state.col6[5] === 'y' ? "displayHolesContainerYellow" : "displayHolesContainerRed"} />
-          <DisplayHoles value={this.state.col7[5]} className={this.state.col7[5] === 'y' ? "displayHolesContainerYellow" : "displayHolesContainerRed"} />
+          <DisplayHoles
```

Comment:
prettier at work here doing it's thing

many developers use prettier, so that when they review each others code:
the noise is removed, and 0% of the effort is focused on how your code is formatted because it's all sorted out for you!

that means if we both use prettier, then in future, we wont get distracted by random code style changes that technically do nothing 😅